rfcs/2022-05-17-11532-chronicle-sink.md

```diff
@@ -0,0 +1,239 @@
+# RFC 11532 - 2022-05-17 - Google Chronicle Sink
+
+This RFC proposes a new sink that sends data into
+[Google Chronicle](https://cloud.google.com/chronicle/docs).
+Chronicle is a Security Information and Event Managent (SIEM) platform to store,
+aggregate and search security telemetry.
+
+## Context
+
+- https://github.com/vectordotdev/vector/issues/11532
+
+## Cross cutting concerns
+
+- The schema support required for the UDM events relies on
+  [this PR](https://github.com/vectordotdev/vector/pull/11743)
+  being merged.
+
+## Scope
+
+### In scope
+
+- A sink that sends data into Google Chronicle. This includes UDM events and
+  unstructured log events.
+
+### Out of scope
+
+- Adding any explicit knowledge of SIEM events as a separate event
+  type into Vector will not be considered.
+
+## Pain
+
+## Proposal
+
+### User Experience
+
+This adds a new sink to Vector to send data into Google Chronicle.
+
+There is a choice of two different types of events that can be sent to Chronicle.
+
+Which type to send will be configured with an `event_type` configuration option.
+
+```toml
+event_type = "unstructured"
+```
+
+or
+
+```toml
+event_type = "udm"
+```
```

Comment:
Eh, I dislike this option. This seems better solved with a schemas option:

```toml
[sinks.chronicle]
type = "gcp_chronicle"
schemas.type = "gcp_udm"
```

Reply:
Ah, ok. Then I think there is a case for splitting up these sinks, right? (i.e., `gcp_chronicle_json`, `gcp_chronicle_udm`). This is similar to how we have different Prometheus sinks for different protocol implementations. Curious what you think @StephenWakely and @jszwedko . I tried to cover the boundaries [here](https://github.com/vectordotdev/vector/blob/master/docs/USER_EXPERIENCE_DESIGN.md#source--sink-boundaries), but it doesn't cover these types of cases.

Reply:
That does make sense to me, especially if we start adding configurable "schema" properties, which wouldn't make sense for the "unstructured" variant (I would name it `gcp_chronicle_unstructured`).

Reply:
Cool. Yeah that was one of the outstanding questions. I'll update the RFC.